templates/coderd.yaml

```diff
@@ -194,46 +179,74 @@ spec:
             initialDelaySeconds: 10
             failureThreshold: 7
             periodSeconds: 10
-{{- if hasKey .Values "cemanager" }}
-{{- include "coder.resources" .Values.cemanager.resources | indent 10 }}
-{{- else }}
-{{- include "coder.resources" .Values.coderd.resources | indent 10 }}
-{{- end }}
+{{- include "coder.resources" (fromYaml (include "movedValue" (dict "Values" .Values "Key" "coderd.resources"))) | indent 10 }}
 {{- include "coder.volumeMounts" . | indent 10 }}
 {{- include "coder.volumes" . | indent 6 }}
+{{- if .Values.coderd.replica.enable }}
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: coder-replica-private-key
+  namespace: {{ .Release.Namespace | quote }}
+type: Opaque
+# Adapted from https://stackoverflow.com/a/64325744
+data:
+  {{- if (lookup "v1" "Secret" .Release.Namespace "coder-replica-private-key") }}
+  privateKey: {{ (lookup "v1" "Secret" .Release.Namespace "coder-replica-private-key").data.privateKey }}
+  {{- else }}
+  privateKey: {{ genPrivateKey "ecdsa" | b64enc }}
+  {{- end }}
+{{- end }}
+{{ if eq (include "coder.useServiceNext" .) "true" }}
 ---
 apiVersion: v1
 kind: Service
 metadata:
+  {{- if (lookup "v1" "Service" .Release.Namespace "ingress-nginx") }}
+  name: nginx-ingress
+  {{- else }}
   name: {{ include "coder.serviceName" . }}
+  {{- end }}
   namespace: {{ .Release.Namespace | quote }}
 spec:
-  type: {{ .Values.serviceType | quote}}
+{{ toYaml .Values.coderd.serviceSpec | indent 2 }}
   selector:
     coder.deployment: {{ include "coder.serviceName" . }}
   ports:
     - name: tcp-{{ include "coder.serviceName" . }}
-      port: 8080
+      port: 80
+      targetPort: 8080
       protocol: TCP
+      appProtocol: http
     - name: tcp-{{ include "coder.serviceName" . }}-https
-      port: 8443
+      port: 443
```

Comment:
Nice! This means we can use privileged ports without running as root :tada: (This was a problem for some customers with nginx-ingress)